src/Features/Core/Portable/Diagnostics/Service/DocumentAnalysisExecutor.cs

```diff
@@ -20,355 +20,358 @@
-/// </summary>
-internal sealed partial class DocumentAnalysisExecutor
+internal sealed partial class DiagnosticAnalyzerService
 {
-    private readonly DiagnosticAnalyzerService _diagnosticAnalyzerService;
-    private readonly CompilationWithAnalyzersPair? _compilationWithAnalyzers;
-    private readonly bool _logPerformanceInfo;
-    private readonly Action? _onAnalysisException;
-
-    private readonly ImmutableArray<DiagnosticAnalyzer> _compilationBasedProjectAnalyzersInAnalysisScope;
-    private readonly ImmutableArray<DiagnosticAnalyzer> _compilationBasedHostAnalyzersInAnalysisScope;
-
-    private ImmutableDictionary<DiagnosticAnalyzer, DiagnosticAnalysisResult>? _lazySyntaxDiagnostics;
-    private ImmutableDictionary<DiagnosticAnalyzer, DiagnosticAnalysisResult>? _lazySemanticDiagnostics;
-
-    public DocumentAnalysisExecutor(
-        DiagnosticAnalyzerService diagnosticAnalyzerService,
-        DocumentAnalysisScope analysisScope,
-        CompilationWithAnalyzersPair? compilationWithAnalyzers,
-        bool logPerformanceInfo,
-        Action? onAnalysisException = null)
-    {
-        AnalysisScope = analysisScope;
-        _diagnosticAnalyzerService = diagnosticAnalyzerService;
-        _compilationWithAnalyzers = compilationWithAnalyzers;
-        _logPerformanceInfo = logPerformanceInfo;
-        _onAnalysisException = onAnalysisException;
-
-        var compilationBasedProjectAnalyzers = compilationWithAnalyzers?.ProjectAnalyzers.ToImmutableHashSet();
-        _compilationBasedProjectAnalyzersInAnalysisScope = compilationBasedProjectAnalyzers != null
-            ? analysisScope.ProjectAnalyzers.WhereAsArray(compilationBasedProjectAnalyzers.Contains)
-            : [];
-
-        var compilationBasedHostAnalyzers = compilationWithAnalyzers?.HostAnalyzers.ToImmutableHashSet();
-        _compilationBasedHostAnalyzersInAnalysisScope = compilationBasedHostAnalyzers != null
-            ? analysisScope.HostAnalyzers.WhereAsArray(compilationBasedHostAnalyzers.Contains)
-            : [];
-    }
-
-    public DocumentAnalysisScope AnalysisScope { get; }
-
-    public DocumentAnalysisExecutor With(DocumentAnalysisScope analysisScope)
-        => new(_diagnosticAnalyzerService, analysisScope, _compilationWithAnalyzers, _logPerformanceInfo, _onAnalysisException);
-
     /// <summary>
-    /// Return all local diagnostics (syntax, semantic) that belong to given document for the given analyzer by calculating them.
+    /// Executes analyzers on a document for computing local syntax/semantic/additional file diagnostics for a specific <see cref="DocumentAnalysisScope"/>.
     /// </summary>
-    public async Task<IEnumerable<DiagnosticData>> ComputeDiagnosticsInProcessAsync(DiagnosticAnalyzer analyzer, CancellationToken cancellationToken)
+    private sealed partial class DocumentAnalysisExecutor
```

Comment:
Check that the file gets renamed here (eventually).

Reply:
Yes.  Names get cleaned up later.